lua/bufferline/render.lua

```diff
@@ -233,8 +241,10 @@ local function render(update_names)
   -- To prevent the expansion of the last click group
   result = result .. '%0@BufferlineMainClickHandler@'
 
-  if layout.actual_width + 1 <= layout.buffers_width and len(items) > 0 then
-    result = result .. hl('BufferTabpageFill') .. icons.bufferline_separator_inactive
+  result = result .. hl(layout.tabpages_width > 0 and 'BufferTabpageFill' or 'TabLineFill')
```

Comment:
We should use a single `Fill` group.

Reply:
This was changed specifically to address a bug report. See [here](https://github.com/romgrk/barbar.nvim/issues/59#issuecomment-732029962) for more, as well as the relevant commit messages:

* d450962c0b2a392ace577170d0d68b9729a38401
* b22866d87d8f4178ca1304e7649203555084d275

Reply:
I'd rather use a single `Fill` group that wouldn't be `Tabpage` specific: it would be a general-purpose fill group, owned by us. `TabLineFill` should be avoided because we don't control it, we should only use it to get its background color. On my colorscheme currently those two hl groups have different foreground colors; if we want to use both we'd need to ensure that the foreground color of those 2 groups is the same. But as I said we don't own `TabLineFill`. We could make it our general-purpose fill group, but then let's use only that one.

I'm not sure I understand the bug report there, but I see it this way: the final `icon_separator_inactive` is sometimes `BufferTabpageFill`, sometimes `TabLineFill` with the current implementation, which doesn't seem ok. It should always use the same group.